cin_validator/rules/cin2022_23/rule_4180.py

```diff
@@ -35,9 +35,10 @@ def validate(
 
     # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
 
-    valid_gender_codes = [1, 2, 0, 9]
+    valid_gender_codes = [1, 2, 0, 9, "1", "2", "0", "9"]
 
     # <GenderCurrent> (N00097) must be present and valid
+
     failing_indices = df[
         df[GenderCurrent].isna() | (~df[GenderCurrent].isin(valid_gender_codes))
```

Comment:
you could also do 
`~df[GenderCurrent].astype("str").isin(valid_gender_codes)` and delete the integer values of valid_gender_codes.

Reply:
In that way, whatever format the input has, you can be sure that it'll be a string when compared to your string expected values.

use `.astype(str)` - no quotes - if the above line raises an error.

Reply:
ah, I was doing astype int which was causing an error, I'll try int now